src/client/search/categories/LeagueCategory.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+
+class LeagueCategory extends Component {
+  componentDidMount() {
+    $('.ui.dropdown.leagues').dropdown({
+      action: 'nothing',
+    });
+  }
+
+  render() {
+    let yearsLeagues = [];
+    if (this.props.leagues) {
+      if (!this.props.leagues.isFetching) {
+        yearsLeagues = this.props.leagues.seasons;
+      }
+    }
+
+    const isFetchingLoader = <i className="purple icon"><div className="ui active inline loader"></div></i>;
+    const isFetchingSmallLoader = <i className="purple icon"><div className="ui active small inline loader"></div></i>;
+
+    const leagueCountTransitionClassName =
+      this.props.tabYear === this.props.currentYear ?
+      'ui purple horizontal label year league count content transition visible'
+      :
+      'ui purple horizontal label year league count content transition hidden';
+
+    return (
+      <div className="large ui dropdown labeled icon button leagues">
+        {yearsLeagues.length === 0 ? isFetchingLoader : <i className="trophy purple icon"></i>}
+        <span className="text">
+          {
+            yearsLeagues.length === 0 ?
+              isFetchingSmallLoader
+            :
+              <div data-id={this.props.tabYear} className={leagueCountTransitionClassName}>
+                {yearsLeagues.length}
+              </div>
+          }
+          Leagues...
```

Comment:
Was momentarily confused as to why you had a spread operator in text, til I realised it was an ellipsis haha